pkgs/build-support/docker/default.nix

```diff
@@ -501,6 +502,7 @@ rec {
           pure = writeText "${baseName}-config.json" (builtins.toJSON {
             inherit created config;
             architecture = defaultArch;
+            variant = if defaultVariant != "" then "v${defaultVariant}" else null;
```

Comment:
According to the manifest spec, this field is "optional", not "nullable". Let's be safe and use `// optionalAttrs` for this.